src/ddmd/statementsem.d

```diff
@@ -483,6 +490,366 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
         result = fs;
     }
 
+    private static template MakeTupleForeachRet(bool isDecl){
+        static if(isDecl)
+        {
+            alias MakeTupleForeachRet = Dsymbols*;
+        }
+        else
+        {
+            alias MakeTupleForeachRet = void;
+        }
+    }
+
+    MakeTupleForeachRet!isDecl makeTupleForeach(bool isStatic, bool isDecl)(ForeachStatement fs, TupleForeachArgs!(isStatic, isDecl) args)
+    {
+        enum returnEarly = isDecl ? q{ return null; } : q{ result = new ErrorStatement(); return; };
```

Comment:
That does not type check, but I can use a nested function with a `static if` inside if that is better.